File: 백준/Silver/3986. 좋은 단어/좋은 단어.py
n = int(input())

pro = [list(input()) for _ in range(n)]
sol = []
count = 0
for pro2 in pro:
    for i in pro2:
        sol.append(i)
        if len(sol) >= 2:
            if sol[-1] == sol[-2]:
                sol.pop()
                sol.pop()
    if sol == []:
        count += 1
    sol = []
print(count)

File: 백준/Silver/3986. 좋은 단어/좋은 단어.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {
  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

    int num = Integer.parseInt(br.readLine());
    int count = 0;

    for(int i=0; i< num; i++) {
      String[] prob = br.readLine().split("");
      Stack<String> sol = new Stack<>();

      for (String str: prob) {

        if (!sol.isEmpty() && sol.peek().equals(str)) {
          sol.pop();
        } else {
          sol.push(str);
        }

      }
      if(sol.isEmpty()) {
        count++;
      }
    }
    System.out.println(count);
  }
}
